pipeline/steps/tf-plan.yaml

```diff
@@ -1,9 +1,12 @@
+parameters:
+  subscription: ''
+
 steps:
 
-  - task: TerraformCLI@0
-    displayName: Terraform plan
-    inputs:
-      command: plan
-      workingDirectory: $(System.DefaultWorkingDirectory)/infrastructure/terraform/
-      environmentServiceName: $(subscription)
-      commandOptions: '-var-file="environment/$(environment)-hmi.tfvars" -out="tfplan-$(environment)" -var "principal_tenant_id=$(principal_tenant_id)" -var "principal_object_id=$(principal_object_id)" -var "prefix=$(prefix)" -var "product=$(product)" -var "environment=$(environment)" -var "location=$(location)"'
\ No newline at end of file
+- task: TerraformCLI@0
+  displayName: Terraform plan
+  inputs:
+    command: plan
+    workingDirectory: $(System.DefaultWorkingDirectory)/infrastructure/terraform/
+    environmentServiceName: ${{ parameters.subscription }}
+    commandOptions: '-var-file="environment/$(environment)-hmi.tfvars" -out="tfplan-$(environment)" -var "principal_tenant_id=$(principal_tenant_id)" -var "principal_object_id=$(principal_object_id)" -var "prefix=$(prefix)" -var "product=$(product)" -var "environment=$(environment)" -var "location=$(location)"'
```

Comment:
this runtime params should be compile params and passed in as runtime in the root pipeline